Source/SwiftLintFramework/Extensions/Array+SwiftLint.swift

```diff
@@ -24,3 +24,26 @@ extension Array where Element: NSTextCheckingResult {
         return map { $0.range }
     }
 }
+
+extension Array where Element: Equatable {
+    var unique: [Element] {
+        var uniqueValues: [Element] = []
+        forEach { item in
+            if !uniqueValues.contains(item) {
+                uniqueValues += [item]
+            }
+        }
+        return uniqueValues
+    }
+}
+
+public extension SequenceType {
+    func categorise<U: Hashable>(@noescape by transform: Generator.Element -> U) -> [U:[Generator.Element]] {
+        var dictionary: [U:[Generator.Element]] = [:]
```

Comment:
we're in the process of adding a new rule requiring a space after the colon in dictionaries: `var dictionary = [U: [Generator.Element]]()`